scripts/skills/perks/perk_rf_the_rush_of_battle.nut

```diff
@@ -1,6 +1,9 @@
 this.perk_rf_the_rush_of_battle <- ::inherit("scripts/skills/skill", {
 	m = {
-		Stacks = 0,
+		// We use two pools because each pool is tracked separately and expires at the end of the next turn
+		StackPool1 = 0,
+		StackPool2 = 0,
+		CurrPool = true, // Which pool is currently gaining stacks. true is pool 1 and false is pool 2. Flipped onTurnEnd.
```

Comment:
Have you considered an approach where `StackPool1` is always the pool that expires next, and `StackPool2` is the one that expires one turn later?
`addStacks` always adds to `StackPool2`
`onTurnStart()` copies `StackPool2` into `StackPool1` and then sets `StackPool2` to 0

Then there is no need for `CurrPool` flip flop

Reply:
That is a great suggestion, thanks. Will do it like that.